src/modules/restApi/adapters/secondary/nextJsRestApiAdapter.ts

```diff
@@ -0,0 +1,152 @@
+import { type RestApiPort } from "@/modules/restApi/ports/secondary/restApiPort";
+import type {
+  DeleteParams,
+  PatchParams,
+  PostParams,
+  UnauthPostParams,
+  GetParams,
+} from "@/modules/restApi/application/entities/restApiParams";
+import { type RequestOptions } from "@/modules/restApi/application/entities/requestOptions";
+
+type NextJsAuthRequestOptions = Required<
```

Comment:
Could we handle all these types and interfaces in a separate file?

Reply:
My general rule of thumb is to keep types that are local in the relevant file, and global types in its own global file. The types here are only gunna be used by this adapter so I think it's fine to just keep it in this file

Reply:
I'm not a fan of this approach; I prefer keeping types or interfaces in separate files to make management easier, but it's up to you